tests/test_ctadata.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+import tempfile
+import os
+from conftest import upstream_webdav_server, hash_file, generate_random_file
+import ctadata
+
+
+@pytest.mark.timeout(30)
+def test_apiclient_list(testing_download_service):
+    with upstream_webdav_server():
+        r = ctadata.list_dir(
+            "", downloadservice=testing_download_service['url'])
+        print(r)
+
+        ctadata.APIClient.downloadservice = testing_download_service['url']
+
+        with pytest.raises(ctadata.api.StorageException):
+            ctadata.list_dir("fake-temp")
+
+        r = ctadata.list_dir("lst/")
+
+        expected = ['lst/', 'lst/users/']
+
+        assert len(r) == 2
+        for entry in r:
+            assert entry['href'] in expected
+
+
+@pytest.mark.timeout(30)
+def test_apiclient_fetch(testing_download_service, caplog):
+    with upstream_webdav_server():
+        ctadata.APIClient.downloadservice = testing_download_service['url']
+
+        r = ctadata.list_dir("lst")
+
+        for entry in r:
+            print(entry)
+            if entry['type'] == 'file' and int(entry['size']) > 10000:
```

Comment:
where does this file come from? Is it generated with fresh `testing_download_service` or remains from another test?

Reply:
It's the content of the original test, I will add a file generation to ensure that the test asserts the download.